diagnostics/model_test/verify_model.py

```diff
@@ -0,0 +1,161 @@
+import tensorflow as tf
+import argparse
+import numpy as np
+import ngraph
+import json
+import os
+
+
+def calculate_output(param_dict, select_device, input_example):
+    """Calculate the output of the imported frozen graph given the input.
+
+    Load the graph def from frozen_graph_file on selected device, then get the tensors based on the input and output name from the graph,
+    then feed the input_example to the graph and retrieves the output vector.
+
+    Args:
+    param_dict: The dictionary contains all the user-input data in the json file.
+    select_device: "NGRAPH" or "CPU".
+    input_example: Random generated input or actual image.
+
+    Returns:
+        The output vector obtained from running the input_example through the graph.
+    """
+    frozen_graph_filename = param_dict["frozen_graph_location"]
+    input_tensor_name = param_dict["input_tensor_name"]
+    output_tensor_name = param_dict["output_tensor_name"]
+
+    with tf.gfile.GFile(frozen_graph_filename, "rb") as f:
+        graph_def = tf.GraphDef()
+        graph_def.ParseFromString(f.read())
```

Comment:
This currently only loads .pb files and not .pbtxt.

You can use ```load_file(graph_file, input_binary, modifier_function_list=[])``` function in ngtf_graph_viewer.py.
for input_binary u can pass: frozen_graph_filename.split('.')[1]=='pb'

Reply:
The script now only supports loading frozen graph .pb file, I think .pbtxt will only contain graph structure but won't contain weights information right, unless you also supply a checkpoint file.

I am plan also to support .pbtxt + checkpoint combination, what do you think?

Reply:
General question - what topologies we tested this on?

Reply:
frozen graphs come in 2 varieties. Both are 'executable':
.pb are binary files and not very human readable
.pbtxt are human readable. .pbtxts can contain weights and constants though.

checkout the 'text or binary' section here
https://www.tensorflow.org/extend/tool_developers/#text_or_binary

However there may also be stripped .pbtxts, which do not have weights etc. Those we wouldn't be able to process I think.